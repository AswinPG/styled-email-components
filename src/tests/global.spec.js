import React from 'react';
import { renderToStaticMarkup } from 'react-dom/server';
import { injectGlobal, ServerStyleSheet } from '../';

describe('global', () => {
  const Document = props => {
    injectGlobal`
      html {
        color: ${props.color};
      }
    `;
    return <html {...props} />;
  };

  it('works with `injectGlobal`', () => {
    const sheet = new ServerStyleSheet();
    renderToStaticMarkup(sheet.collectStyles(<Document color="blue" />));
    expect(sheet.getStyleTags()).toMatchSnapshot();
  });
});
