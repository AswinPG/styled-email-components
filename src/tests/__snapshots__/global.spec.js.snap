// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`global works with \`injectGlobal\` 1`] = `
"<style data-styled-components=\\"\\">
/* sc-component-id: sc-global-2614879494 */
html{color:blue;}</style>"
`;
